### ISSUE TITLE
Versionierung

### ISSUE THREAD

**Issue by sheepyhollow** [ClockGU/features#16]
Haben wir ein Benennungsschema und einen Dienst, der die Versionsnummer managt?

**Comment by sheepyhollow** [comment]
Heißt: Versionierung für die API und nicht fürs Frontend?

**Comment by sheepyhollow** [comment]
Wir sind uns aber einig, dass wir keine namentliche Abgrenzung zum alten Clock brauchen? Intern reden wir ja manchmal von v1 und v2.

**Comment by mimischi** [comment]
> Wir sind uns aber einig, dass wir keine namentliche Abgrenzung zum alten Clock brauchen? Intern reden wir ja manchmal von v1 und v2.

Ja, ich würde `django-clock` als "nicht existent" behandeln und das neue Produkt einfach bei 1.0 anfangen.

> Tatsächlich habe ich über genau das auch scon nachgedacht. Wenn es nur um das Backend geht habe ich mal gelesen, dass sollte man eine öffentlich API anbieten (gerade wenn man diese nur kostenpflichtig zur Verfügung stellt), welche von anderen APPs/Produkten benutzt wird sollte man unbedingt Versionieren. Grund dafür ist, dass wenn man bspw. die Datenstruktur eines Endpoints ändert eine neue Version via v2.api.example.de bereitzustellen, da diese Änderung in anderen APPs Fehler verusrsachen könnte.

Die Frage ist eben, können wir die URL Struktur so ändern, dass die "zwei Versionen" der API immer innerhalb einer Django App (und eines Backends) laufen? Das würde natürlich bedeuten, dass wir nach außen nie Business-Logik ändern dürfen, die mit Models zu tun hat.

Schön wären langfristig also URLs wie:
- `api.clock.uni-frankfurt.de/v1/`
- `api.clock.uni-frankfurt.de/v2/`

Aktuell läuft im Backend ja explizitg der `/api/` Pfad mit, ist meiner Meinung nach aber überflüssig. Das Backend ist ja eine reine API.

**Comment by sheepyhollow** [comment]
Hieraus sollten wir genaue To-Dos bauen.
Ich würde argumentieren, dass wir zumindest zum angepeilten Release-Date (Mitte März) eine Entscheidung haben sollten, wie wir die Versionierung handhaben.

**Comment by sheepyhollow** [comment]
Wie relevant ist das denn? Ich würde das ansonsten aus den Open-Beta Milestone herausnehmen.

**Comment by mimischi** [comment]
Ja lass das mal rausnehmen.

**Comment by sheepyhollow** [comment]
Haben wir diese Frage eigentlich geklärt?
